Fix Helix testing issues with stable versions and local builds

dougbu opened the pull request:
- include all shipping packages in Helix payloads that need runtime
  - remove hard-coded `-ci` that broke Helix tests with stable versions or local builds
  - for local builds, do not assume `$(Configuration)` is Release
- ensure Helix.props is imported in all cases that Helix.targets is
  - discovered because `$(TestDependsOnAspNetRuntime)` wasn't defined in two `*.npmproj` projects
- support `$(HelixTargetQueues)` property used in RunHelix.ps1
  - lost somewhere along the lines; script ran full matrix

nits:
- clean up redundant addition of runtime and ref/ packages
  - `@(HelixContent)` additions in `_CreateHelixWorkItem` target are ignored

dougbu committed:
Fix Helix testing issues with stable versions and local builds
- include all shipping packages in Helix payloads that need runtime
  - remove hard-coded `-ci` that broke Helix tests with stable versions or local builds
  - for local builds, do not assume `$(Configuration)` is Release
- ensure Helix.props is imported in all cases that Helix.targets is
  - discovered because `$(TestDependsOnAspNetRuntime)` wasn't defined in two `*.npmproj` projects
- support `$(HelixTargetQueues)` property used in RunHelix.ps1
  - lost somewhere along the lines; script ran full matrix

nits:
- clean up redundant addition of runtime and ref/ packages
  - `@(HelixContent)` additions in `_CreateHelixWorkItem` target are ignored

dougbu commented on Directory.Build.props:
```diff
@@ -13,7 +13,7 @@
     <IsBenchmarkProject Condition="$(MSBuildProjectName.EndsWith('.Performance')) OR $(RepoRelativeProjectDir.Contains('perf')) OR $(RepoRelativeProjectDir.Contains('benchmarkapps'))">true</IsBenchmarkProject>
     <IsSpecificationTestProject Condition="$(MSBuildProjectName.EndsWith('.Specification.Tests'))">true</IsSpecificationTestProject>
-    <IsUnitTestProject>false</IsUnitTestProject>
+    <IsUnitTestProject Condition=" '$(IsUnitTestProject)' == '' ">false</IsUnitTestProject>
```
See comments in PR description about ensuring Helix.props is imported. Arcade SDK sets `$(IsTestProject)` based on `$(IsUnitTestProject)` and a few other properties. A couple of `*.npmproj` test projects don't follow our naming conventions and previously set `$(IsTestProject)` _after_ this file was imported.

dougbu commented on eng/helix/helix.proj:
```diff
@@ -11,7 +11,8 @@
 
-  <ItemGroup>
+  <!-- Microsoft.DotNet.Helix.Sdk.MultiQueue.targets splits $(HelixTargetQueues) into @(HelixTargetQueue) items. -->
+  <ItemGroup Condition=" '$(HelixTargetQueues)' == '' ">
```
See comments in PR description about RunHelix.ps1

dougbu commented on eng/targets/Helix.targets:
```diff
@@ -16,8 +16,14 @@
-    <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\*-ci.nupkg" />
+  <ItemGroup Condition=" $(TestDependsOnAspNetRuntime) ">
+    <!-- Grab all shipping packages. -->
+    <HelixContent Include="$(RepoRoot)artifacts\packages\$(Configuration)\Shipping\*$(SharedFxVersion).nupkg" />
```
Broadening the wildcard ensures items are added in (local) Debug builds and after stabilizing versions.

dougbu commented on eng/targets/Helix.targets:
```diff
@@ -101,19 +104,6 @@ Usage: dotnet msbuild /t:Helix src/MyTestProject.csproj
-    </ItemGroup>
-
-    <ItemGroup Condition=" '$(TestDependsOnAspNetRuntime)' == 'true' ">
-      <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\Microsoft.AspNetCore.App.Runtime.win-x64.$(SharedFxVersion).nupkg" />
```
These `@(HelixContent)` items were ignored because they aren't available when the `Publish` target executes in the inner `_PublishHelixArchive` build; items don't cross `<MSBuild />` boundaries.

dougbu reviewed (commented):
A fun onion peeled 😺

HaoK reviewed (approved):
Maybe more like a blooming onion, deep fried and tasty in moderation, yet super unhealthy... msbuild quirks fun...

dougbu committed:
!fixup! Correct Java and NPM projects I broke
- Java projects don't use xUnit
- NPM projects I changed in previous commit lack `Publish` target

dougbu committed:
!fixup! Undo changes in NPM projects
- most recent changes break non-Helix testing of these projects
- instead, get Helix.targets working again when Helix.props isn't imported

nit: mention '+' separation of `-HelixQueues` argument to RunHelix.ps1

dougbu commented on eng/scripts/RunHelix.ps1:
```diff
@@ -6,7 +6,7 @@
     The test project to publish and send to Helix.
 .PARAMETER HelixQueues
-    Set the Helix queues to use, the list is ';' separated.
+    Set the Helix queues to use. The list is '+' or ';'-separated.
```
Supposedly Microsoft.DotNet.Helix.Sdk.MultiQueue.targets supports ','-separation but that doesn't seem to work
